sources/cfsignal/src/cloudformation.rs

```diff
@@ -0,0 +1,65 @@
+use crate::error::{self, Result};
+use imdsclient::ImdsClient;
+use log::info;
+use rusoto_cloudformation::{
+    CloudFormation, CloudFormationClient, SignalResourceInput,
+};
+use rusoto_core::Region;
+use snafu::OptionExt;
+use snafu::ResultExt;
+use std::str::FromStr;
+
+pub(crate) struct SignalResource {}
+
+impl SignalResource {
+    #[tokio::main]
+    pub(crate) async fn signal_resource(
+        stack_name: String,
+        logical_resource_id: String,
+        status: String,
+    ) -> Result<()> {
+        info!("Connecting to IMDS");
+        let mut client = ImdsClient::new().await.context(error::ImdsClient)?;
+        let instance_id = get_instance_id(&mut client).await.unwrap();
+        let region = get_region(&mut client).await.unwrap();
+
+        info!("Region: {:?} - InstanceID: {:?} - Signal: {:?}", region, instance_id, status);
+
+        let client = CloudFormationClient::new(Region::from_str(&region).unwrap());
+        let signal_resource_input = SignalResourceInput {
+            stack_name: stack_name,
+            logical_resource_id: logical_resource_id,
+            status: status,
+            unique_id: instance_id,
+        };
+
+        match client.signal_resource(signal_resource_input).await {
+            Ok(_output) => (),
+            Err(error) => {
+                println!("Error: {:?}", error);
+            }
+        }
+
+        Ok(())
```

Comment:
This should be collapsed down to just:
```
client.signal_resource(signal_resource_input).await.context(error::SignalResource)
```

If we want to just print the error and exit zero, we can do that in `main` - our standard boilerplate for this is:
```
#[tokio::main]
async fn main() {
    if let Err(e) = run().await {
        eprintln!("{}", e);
        // could make this `0` if we really want to ignore errors
        process::exit(1); 
    }
}
```